### ISSUE TITLE
`LoadError: no unique matching method found for the specified argument types` during precompilation

### ISSUE THREAD

**Issue by IanButterworth** [MakieOrg/Makie.jl#2704]
```
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.10.0-DEV.649 (2023-02-22)
 _/ |\__'_|_|_|\__'_|  |  Commit e3d366f1966 (0 days old master)
|__/                   |


(@v1.10) pkg> st
Status `~/.julia/environments/v1.10/Project.toml`
  [e9467ef8] GLMakie v0.8.2

julia> using GLMakie
[ Info: Precompiling GLMakie [e9467ef8-e4e7-5192-8a1a-b1aee30e663a]
ERROR: LoadError: no unique matching method found for the specified argument types
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] _which(tt::Type; method_table::Nothing, world::UInt64, raise::Bool)
    @ Base ./reflection.jl:1576
  [3] _which
    @ ./reflection.jl:1563 [inlined]
  [4] which(tt::Any)
    @ Base ./reflection.jl:1602
  [5] which
    @ ./reflection.jl:1593 [inlined]
  [6] is_propertynames_overloaded
    @ ~/.julia/packages/ConstructionBase/ZLOSb/src/ConstructionBase.jl:52 [inlined]
  [7] #s26#2
    @ ~/.julia/packages/ConstructionBase/ZLOSb/src/ConstructionBase.jl:56 [inlined]
  [8] var"#s26#2"(::Any, obj::Any)
    @ ConstructionBase ./none:0
  [9] (::Core.GeneratedFunctionStub)(::UInt64, ::LineNumberNode, ::Any, ::Vararg{Any})
    @ Core ./boot.jl:600
 [10] setproperties_object(obj::Makie.GlyphInfo, patch::NamedTuple{(:origin,), Tuple{GeometryBasics.Point{2, Float32}}})
    @ ConstructionBase ~/.julia/packages/ConstructionBase/ZLOSb/src/ConstructionBase.jl:186
 [11] setproperties
    @ ~/.julia/packages/ConstructionBase/ZLOSb/src/ConstructionBase.jl:105 [inlined]
 [12] set
    @ ~/.julia/packages/Setfield/PdKfV/src/lens.jl:122 [inlined]
 [13] set
    @ ~/.julia/packages/Setfield/PdKfV/src/lens.jl:189 [inlined]
 [14] modify
    @ ~/.julia/packages/Setfield/PdKfV/src/lens.jl:106 [inlined]
 [15] macro expansion
    @ ~/.julia/packages/Setfield/PdKfV/src/sugar.jl:204 [inlined]
 [16] apply_lineheight!(lines::Vector{Vector{Makie.GlyphInfo}}, lh::Float64)
    @ Makie ~/.julia/packages/Makie/gAmAB/src/basic_recipes/text.jl:366
 [17] layout_text(rt::Makie.RichText, ts::Float32, f::FreeTypeAbstraction.FTFont, fset::MakieCore.Attributes, al::Tuple{Symbol, Symbol}, rot::Makie.Quaternion{Float64}, jus::MakieCore.Automatic, lh::Float64, col::ColorTypes.RGBA{Float32})
    @ Makie ~/.julia/packages/Makie/gAmAB/src/basic_recipes/text.jl:351
 [18] _get_glyphcollection_and_linesegments(rt::Makie.RichText, index::Int64, ts::Float32, f::FreeTypeAbstraction.FTFont, fset::MakieCore.Attributes, al::Tuple{Symbol, Symbol}, rot::Makie.Quaternion{Float64}, jus::MakieCore.Automatic, lh::Float64, col::ColorTypes.RGBA{Float32}, scol::ColorTypes.RGBA{Float32}, swi::Int64, www::Int64)
    @ Makie ~/.julia/packages/Makie/gAmAB/src/basic_recipes/text.jl:302
 [19] call_composed
    @ ./operators.jl:1035 [inlined]
 [20] call_composed
    @ ./operators.jl:1034 [inlined]
 [21] (::ComposedFunction{Makie.var"#push_args#1885"{Vector{Int64}, Vector{ColorTypes.RGBA{Float32}}, Vector{Float32}, Vector{GeometryBasics.Point{2, Float32}}, Vector{Makie.GlyphCollection}}, typeof(Makie._get_glyphcollection_and_linesegments)})(::Makie.RichText, ::Vararg{Any}; kw::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./operators.jl:1031
 [22] (::ComposedFunction{Makie.var"#push_args#1885"{Vector{Int64}, Vector{ColorTypes.RGBA{Float32}}, Vector{Float32}, Vector{GeometryBasics.Point{2, Float32}}, Vector{Makie.GlyphCollection}}, typeof(Makie._get_glyphcollection_and_linesegments)})(::Makie.RichText, ::Vararg{Any})
    @ Base ./operators.jl:1031
 [23] macro expansion
    @ ~/.julia/packages/Makie/gAmAB/src/utilities/utilities.jl:202 [inlined]
 [24] broadcast_foreach(::ComposedFunction{Makie.var"#push_args#1885"{Vector{Int64}, Vector{ColorTypes.RGBA{Float32}}, Vector{Float32}, Vector{GeometryBasics.Point{2, Float32}}, Vector{Makie.GlyphCollection}}, typeof(Makie._get_glyphcollection_and_linesegments)}, ::Vector{AbstractString}, ::UnitRange{Int64}, ::Float32, ::FreeTypeAbstraction.FTFont, ::MakieCore.Attributes, ::Tuple{Symbol, Symbol}, ::Makie.Quaternion{Float64}, ::MakieCore.Automatic, ::Float64, ::ColorTypes.RGBA{Float32}, ::ColorTypes.RGBA{Float32}, ::Int64, ::Int64)
    @ Makie ~/.julia/packages/Makie/gAmAB/src/utilities/utilities.jl:188
 [25] (::Makie.var"#1881#1884"{Base.RefValue{Vector{Int64}}, Observables.Observable{Vector{ColorTypes.RGBA{Float32}}}, Observables.Observable{Vector{Float32}}, Observables.Observable{Vector{GeometryBasics.Point{2, Float32}}}, Observables.Observable{Vector{Makie.GlyphCollection}}})(str::Vector{AbstractString}, ts::Float64, f::Symbol, fs::MakieCore.Attributes, al::Tuple{Symbol, Symbol}, rot::Float64, jus::MakieCore.Automatic, lh::Float64, col::ColorTypes.RGBA{Float32}, scol::Tuple{Symbol, Float64}, swi::Int64, www::Int64)
    @ Makie ~/.julia/packages/Makie/gAmAB/src/basic_recipes/text.jl:37
 [26] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./essentials.jl:828
 [27] invokelatest(::Any, ::Any, ::Vararg{Any})
    @ Base ./essentials.jl:825
 [28] (::Observables.OnAny)(value::Any)
    @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:415
 [29] #invokelatest#2
    @ ./essentials.jl:828 [inlined]
 [30] invokelatest
    @ ./essentials.jl:825 [inlined]
 [31] notify(observable::Observables.AbstractObservable)
    @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169
 [32] plot!(plot::MakieCore.Text{Tuple{Vector{GeometryBasics.Point{3, Float32}}}})
    @ Makie ~/.julia/packages/Makie/gAmAB/src/basic_recipes/text.jl:65
 [33] plot!
    @ ~/.julia/packages/Makie/gAmAB/src/interfaces.jl:423 [inlined]
 [34] plot!(scene::MakieCore.Text{Tuple{Vector{Tuple{AbstractString, GeometryBasics.Point{2, Float32}}}}}, P::Type{MakieCore.Text}, attributes::MakieCore.Attributes, args::Observables.Observable{Vector{GeometryBasics.Point{3, Float32}}}; kw_attributes::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Makie ~/.julia/packages/Makie/gAmAB/src/interfaces.jl:328
 [35] plot!
    @ ~/.julia/packages/Makie/gAmAB/src/interfaces.jl:295 [inlined]
 [36] #plot!#162
    @ ~/.julia/packages/Makie/gAmAB/src/interfaces.jl:279 [inlined]
 [37] text!(::MakieCore.Text{Tuple{Vector{Tuple{AbstractString, GeometryBasics.Point{2, Float32}}}}}, ::Vararg{Any}; attributes::Base.Pairs{Symbol, Any, NTuple{23, Symbol}, NamedTuple{(:text, :transparency, :strokecolor, :visible, :space, :fonts, :ssao, :word_wrap_width, :justification, :font, :align, :strokewidth, :rotation, :overdraw, :lineheight, :transformation, :model, :depth_shift, :color, :offset, :markerspace, :fontsize, :inspectable), Tuple{Observables.Observable{Vector{AbstractString}}, Observables.Observable{Any}, Observables.Observable{Any}, Observables.Observable{Any}, Observables.Observable{Any}, MakieCore.Attributes, Vararg{Observables.Observable{Any}, 17}}}})
    @ MakieCore ~/.julia/packages/MakieCore/LOKNz/src/recipes.jl:38
 [38] plot!(plot::MakieCore.Text{Tuple{Vector{Tuple{AbstractString, GeometryBasics.Point{2, Float32}}}}})
    @ Makie ~/.julia/packages/Makie/gAmAB/src/basic_recipes/text.jl:152
 [39] plot!(scene::Makie.Scene, P::Type{MakieCore.Text{Tuple{Vector{Tuple{AbstractString, GeometryBasics.Point{2, Float32}}}}}}, attributes::MakieCore.Attributes, input::Tuple{Observables.Observable{Vector{Tuple{AbstractString, GeometryBasics.Point{2, Float32}}}}}, args::Observables.Observable{Tuple{Vector{Tuple{AbstractString, GeometryBasics.Point{2, Float32}}}}})
    @ Makie ~/.julia/packages/Makie/gAmAB/src/interfaces.jl:413
 [40] plot!(scene::Makie.Scene, P::Type{MakieCore.Text}, attributes::MakieCore.Attributes, args::Observables.Observable{Vector{Tuple{AbstractString, GeometryBasics.Point{2, Float32}}}}; kw_attributes::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Makie ~/.julia/packages/Makie/gAmAB/src/interfaces.jl:328
 [41] plot!
    @ ~/.julia/packages/Makie/gAmAB/src/interfaces.jl:295 [inlined]
 [42] #plot!#162
    @ ~/.julia/packages/Makie/gAmAB/src/interfaces.jl:279 [inlined]
 [43] text!(::Makie.Scene, ::Vararg{Any}; attributes::Base.Pairs{Symbol, Any, NTuple{8, Symbol}, NamedTuple{(:align, :rotation, :fontsize, :font, :color, :visible, :markerspace, :inspectable), Tuple{Observables.Observable{Tuple{Symbol, Symbol}}, Observables.Observable{Any}, Observables.Observable{Any}, Observables.Observable{Any}, Observables.Observable{Any}, Observables.Observable{Any}, Symbol, Bool}}})
    @ MakieCore ~/.julia/packages/MakieCore/LOKNz/src/recipes.jl:38
 [44] Makie.LineAxis(parent::Makie.Scene, attrs::MakieCore.Attributes)
    @ Makie ~/.julia/packages/Makie/gAmAB/src/makielayout/lineaxis.jl:473
 [45] Makie.LineAxis(parent::Makie.Scene; kwargs::Base.Pairs{Symbol, V, Tuple{Vararg{Symbol, N}}, NamedTuple{names, T}} where {V, N, names, T<:Tuple{Vararg{Any, N}}})
    @ Makie ~/.julia/packages/Makie/gAmAB/src/makielayout/lineaxis.jl:3
 [46] initialize_block!(ax::Makie.Axis; palette::Nothing)
    @ Makie ~/.julia/packages/Makie/gAmAB/src/makielayout/blocks/axis.jl:316
 [47] initialize_block!(ax::Makie.Axis)
    @ Makie ~/.julia/packages/Makie/gAmAB/src/makielayout/blocks/axis.jl:166
 [48] _block(::Type{Makie.Axis}, ::Makie.Figure; bbox::Nothing, kwargs::Base.Pairs{Symbol, typeof(log10), Tuple{Symbol, Symbol}, NamedTuple{(:yscale, :xscale), Tuple{typeof(log10), typeof(log10)}}})
    @ Makie ~/.julia/packages/Makie/gAmAB/src/makielayout/blocks.jl:405
 [49] _block
    @ ~/.julia/packages/Makie/gAmAB/src/makielayout/blocks.jl:295 [inlined]
 [50] #_#1085
    @ ~/.julia/packages/Makie/gAmAB/src/makielayout/blocks.jl:276 [inlined]
 [51] plot(P::Type{MakieCore.Heatmap}, args::Matrix{Float64}; axis::NamedTuple{(:yscale, :xscale), Tuple{typeof(log10), typeof(log10)}}, figure::NamedTuple{(), Tuple{}}, kw_attributes::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Makie ~/.julia/packages/Makie/gAmAB/src/figureplotting.jl:50
 [52] plot
    @ ~/.julia/packages/Makie/gAmAB/src/figureplotting.jl:31 [inlined]
 [53] #heatmap#19
    @ ~/.julia/packages/MakieCore/LOKNz/src/recipes.jl:34 [inlined]
 [54] macro expansion
    @ ~/.julia/packages/Makie/gAmAB/precompile/shared-precompile.jl:47 [inlined]
 [55] top-level scope
    @ ~/.julia/packages/Makie/gAmAB/src/precompiles.jl:6
 [56] include(mod::Module, _path::String)
    @ Base ./Base.jl:467
 [57] include
    @ ~/.julia/packages/Makie/gAmAB/src/Makie.jl:1 [inlined]
 [58] macro expansion
    @ ~/.julia/packages/Makie/gAmAB/src/precompiles.jl:15 [inlined]
 [59] macro expansion
    @ ~/.julia/packages/SnoopPrecompile/1XXT1/src/SnoopPrecompile.jl:62 [inlined]
 [60] top-level scope
    @ ~/.julia/packages/Makie/gAmAB/src/precompiles.jl:12
 [61] include(mod::Module, _path::String)
    @ Base ./Base.jl:467
 [62] include(x::String)
    @ Makie ~/.julia/packages/Makie/gAmAB/src/Makie.jl:1
 [63] top-level scope
    @ ~/.julia/packages/Makie/gAmAB/src/Makie.jl:330
 [64] include
    @ ./Base.jl:467 [inlined]
 [65] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::String)
    @ Base ./loading.jl:2035
in expression starting at /Users/ian/.julia/packages/Makie/gAmAB/precompile/shared-precompile.jl:46
in expression starting at /Users/ian/.julia/packages/Makie/gAmAB/src/precompiles.jl:11
in expression starting at /Users/ian/.julia/packages/Makie/gAmAB/src/Makie.jl:1
in expression starting at stdin:2
ERROR: LoadError: Failed to precompile Makie [ee78f7c6-11fb-53f2-987a-cfe4a2b5a57a] to "/Users/ian/.julia/compiled/v1.10/Makie/jl_ShPT0J".
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, keep_loaded_modules::Bool)
    @ Base ./loading.jl:2285
  [3] compilecache
    @ ./loading.jl:2152 [inlined]
  [4] _require(pkg::Base.PkgId, env::String)
    @ Base ./loading.jl:1796
  [5] _require_prelocked(uuidkey::Base.PkgId, env::String)
    @ Base ./loading.jl:1654
  [6] macro expansion
    @ ./loading.jl:1640 [inlined]
  [7] macro expansion
    @ ./lock.jl:267 [inlined]
  [8] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:1603
  [9] include
    @ ./Base.jl:467 [inlined]
 [10] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
    @ Base ./loading.jl:2035
 [11] top-level scope
    @ stdin:2
in expression starting at /Users/ian/.julia/packages/GLMakie/N16Fq/src/GLMakie.jl:1
in expression starting at stdin:2
ERROR: Failed to precompile GLMakie [e9467ef8-e4e7-5192-8a1a-b1aee30e663a] to "/Users/ian/.julia/compiled/v1.10/GLMakie/jl_PXqfze".
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, keep_loaded_modules::Bool)
   @ Base ./loading.jl:2285
 [3] compilecache
   @ ./loading.jl:2152 [inlined]
 [4] _require(pkg::Base.PkgId, env::String)
   @ Base ./loading.jl:1796
 [5] _require_prelocked(uuidkey::Base.PkgId, env::String)
   @ Base ./loading.jl:1654
 [6] macro expansion
   @ ./loading.jl:1640 [inlined]
 [7] macro expansion
   @ ./lock.jl:267 [inlined]
 [8] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1603
```

**Comment by SimonDanisch** [comment]
I guess that should be an issue for ConstructionBase.jl...
And we should maybe consider, if we really need `Setfield.jl` in Makie....

**Comment by jw3126** [comment]
> I guess that should be an issue for ConstructionBase.jl... And we should maybe consider, if we really need `Setfield.jl` in Makie....

I am curious, is Setfield.jl a source of trouble for you? Because we receive extremely few bug reports, my assumption was that it is very stable.

**Comment by SimonDanisch** [comment]
I'm not saying it's unstable, just that its a really good idea for Makie's maintenance & precompile times to rely on less packages... And I guess it seems to (at least indirectly) hook into some internals that have changed in 1.10 ;)

**Comment by rafaqz** [comment]
This looks like world age issues from from when ConstructionBase.jl swapped from using `setfield` to `setproperties`. Its checking for `propertynames` overloads with `which` behind a generated function. 

@aplavin this might need a rethink. It seems pretty complicated and fragile to me.

**Comment by IanButterworth** [comment]
https://github.com/JuliaObjects/ConstructionBase.jl/issues/72 has been closed

**Comment by IanButterworth** [comment]
That fix is released in ConstructionBase v1.5.0 now
